src/wrapper/executor.rs

```diff
@@ -0,0 +1,55 @@
+use {objects::JObject, JavaVM, JNIEnv, errors::*};
+
+use std::sync::Arc;
+
+/// The capacity of local frames, allocated for attached threads by default. Same as the default
+/// value Hotspot uses when calling native Java methods.
+pub const DEFAULT_LOCAL_FRAME_CAPACITY: i32 = 32;
+
+/// Thread attachment manager. Attaches threads as daemons, hence they do not block
+/// JVM exit. Finished threads detach automatically.
+#[derive(Clone)]
+pub struct Executor {
+    vm: Arc<JavaVM>,
```

Comment:
> Is it correct that Executor is not Send + Sync whilst JavaVM — is? Isn't it appropriate to make Executor such?

JavaVM is Send & Sync --> Arc<JavaVM> is Send & Sync --> Executor is Send & Sync.

Reply:
1. Executor is Send + Sync, it's visible in docs. @skletsun described it correctly
<img width="374" alt="изображение" src="https://user-images.githubusercontent.com/6566674/60014969-e994b100-968a-11e9-83fd-ade166ad398a.png">
2. The main reason of putting `JavaVM` in `Arc` is to make it clonable and avoid dangling pointers. Are there any problems with `Arc` here?
3. It's interesting option. We can do it and it will simplify the code a bit, but probably in a separate PR.

Reply:
I don't know if there are any problems, it just seems somewhat strange to use reference-counting with basically a _pointer_ to a foreign object (= as a pointer it has no Drop implementation, therefore, its atomic destruction, which Arc handles, is not an issue).

Reply:
It's a safe way to clone pointer. Otherwise you can't guarantee that JVM will not be destroyed before the last use of the pointer.